sklearn/inspection/_partial_dependence.py

```diff
@@ -655,10 +655,12 @@ def convert_feature(fx):
 
     features = tmp_features
 
-    if isinstance(ax, list):
-        if len(ax) != len(features):
-            raise ValueError("Expected len(ax) == len(features), "
-                             "got len(ax) = {}".format(len(ax)))
+    # Early exit if the axes does not have the correct number of axes
+    if ax is not None and not isinstance(ax, plt.Axes):
+        axes = check_array(ax, dtype=object, ensure_2d=False)
```

Comment:
Do we need all the check of check_array, or we could just call `np.asarray`?

Reply:
Yeah, I'm also not sure we should use `check_array` to validate the list of matplotlib axis.